src/parcsr_ls/par_mgr_interp.c

```diff
@@ -731,171 +733,159 @@ hypre_ExtendWtoPHost(HYPRE_Int      P_nr_of_rows,
-   hypre_CSRMatrixData(hypre_ParCSRMatrixOffd(P)) = hypre_CSRMatrixData(W_offd);
-   hypre_CSRMatrixJ(W_offd)    = NULL;
-   hypre_CSRMatrixData(W_offd) = NULL;
+                                hypre_CSRMatrixNumNonzeros(W_offd));
+   P_diag = hypre_ParCSRMatrixDiag(P);
+   P_offd = hypre_ParCSRMatrixOffd(P);
 
-   if (method > 0)
-   {
-      hypre_ParCSRMatrixColMapOffd(P)    = hypre_ParCSRMatrixColMapOffd(A_FC);
-      hypre_ParCSRMatrixColMapOffd(P)    = hypre_ParCSRMatrixColMapOffd(A_FC);
-      hypre_ParCSRMatrixColMapOffd(A_FC) = NULL;
-      hypre_ParCSRMatrixColMapOffd(A_FC) = NULL;
-      hypre_ParCSRMatrixNumNonzeros(P)   = hypre_ParCSRMatrixNumNonzeros(A_FC) +
-                                           hypre_ParCSRMatrixGlobalNumCols(A_FC);
-   }
-   else
-   {
-      hypre_ParCSRMatrixNumNonzeros(P) = nC_global;
-   }
+   hypre_CSRMatrixMemoryLocation(P_diag) = memory_location_P;
+   hypre_CSRMatrixMemoryLocation(P_offd) = memory_location_P;
+   hypre_CSRMatrixI(P_diag)              = P_diag_i;
+   hypre_CSRMatrixJ(P_diag)              = P_diag_j;
+   hypre_CSRMatrixData(P_diag)           = P_diag_data;
+   hypre_CSRMatrixI(P_offd)              = P_offd_i;
+   hypre_CSRMatrixJ(P_offd)              = hypre_TAlloc(HYPRE_Int,
+                                                        hypre_CSRMatrixNumNonzeros(W_offd),
+                                                        memory_location_P);
+   hypre_CSRMatrixData(P_offd)           = hypre_TAlloc(HYPRE_Complex,
+                                                        hypre_CSRMatrixNumNonzeros(W_offd),
+                                                        memory_location_P);
+   hypre_ParCSRMatrixColMapOffd(P)       = hypre_TAlloc(HYPRE_BigInt,
+                                                        hypre_CSRMatrixNumCols(W_offd),
+                                                        memory_location_P);
+
+   hypre_TMemcpy(hypre_CSRMatrixJ(P_offd), hypre_CSRMatrixJ(W_offd),
+                 HYPRE_Int, hypre_CSRMatrixNumNonzeros(W_offd),
+                 memory_location_P, memory_location_P);
+
+   hypre_TMemcpy(hypre_CSRMatrixData(P_offd), hypre_CSRMatrixData(W_offd),
+                 HYPRE_Complex, hypre_CSRMatrixNumNonzeros(W_offd),
+                 memory_location_P, memory_location_P);
+
+   hypre_TMemcpy(hypre_ParCSRMatrixColMapOffd(P), hypre_ParCSRMatrixColMapOffd(Wp),
+                 HYPRE_BigInt, hypre_CSRMatrixNumCols(W_offd),
+                 memory_location_P, memory_location_P);
+
+   hypre_ParCSRMatrixSetNumNonzeros(P);
    hypre_ParCSRMatrixDNumNonzeros(P) = (HYPRE_Real) hypre_ParCSRMatrixNumNonzeros(P);
    hypre_MatvecCommPkgCreate(P);
 
    /* Set output pointer */
-   *P_ptr = P;
-
-   /* Free memory */
-   hypre_ParCSRMatrixDestroy(A_FF);
-   hypre_ParCSRMatrixDestroy(A_FC);
-   if (method <= 0)
-   {
-      hypre_CSRMatrixDestroy(W_diag);
-      hypre_CSRMatrixDestroy(W_offd);
-   }
+   *Wp_ptr = Wp;
+   *P_ptr  = P;
+   *Wp_ptr = NULL;
```

Comment:
Why is this set to NULL right after it's set to Wp?

Reply:
Thanks for catching! This was the result of a bad merge